src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -50,11 +50,34 @@ void shape_infer(const Squeeze* op,
         OPENVINO_SUPPRESS_DEPRECATED_END
 
         std::vector<int64_t> axes;
-        if (arg_rank.is_static() && axes_shape.is_static() && get_data_as_int64<T>(1, op, axes, constant_data)) {
-            OPENVINO_SUPPRESS_DEPRECATED_START
-            normalize_axes(op, arg_rank.get_length(), axes);
-            OPENVINO_SUPPRESS_DEPRECATED_END
-            unique_axes.reset(new std::set<int64_t>(axes.cbegin(), axes.cend()));
+        if (arg_rank.is_static() && axes_shape.is_static()) {
+            if (get_data_as_int64<T>(1, op, axes, constant_data)) {
+                OPENVINO_SUPPRESS_DEPRECATED_START
+                normalize_axes(op, arg_rank.get_length(), axes);
+                OPENVINO_SUPPRESS_DEPRECATED_END
+                unique_axes.reset(new std::set<int64_t>(axes.cbegin(), axes.cend()));
+            } else if (axes_shape.size() == 1 && axes_shape[0].get_min_length() > 0) {
+                // The `axes` input is a Parameter
+                output_shape.resize(0);
+                int64_t squeezable_dims_count =
+                    std::count_if(arg_shape.begin(), arg_shape.end(), [&](const DimType& dim) {
+                        if (dim.compatible(1)) {
+                            return true;
+                        } else {
+                            // Copy not squeezable dimensions to the output shape
+                            output_shape.push_back(dim);
+                            return false;
+                        }
+                    });
+                if (squeezable_dims_count > axes_shape[0].get_min_length()) {
+                    // If the number of squeezable (1 or with 1 in range) dimensions is bigger than the number of axes,
+                    // only rank can be deduced (assuming uniqueness of axes)
+                    output_shape = PartialShape::dynamic(arg_shape.size() - shape_size(axes_shape.to_shape()));
```

Comment:
This block tries to get squeeze, try move output shape creation to block below e.g. somewhere between line 123-130

Reply:
I have tried (https://github.com/openvinotoolkit/openvino/pull/17238/commits/34d742885e01a3d98be15c189b24fde92591e6e4), but then some checks like number of inputs and axes shape need to be repeated.
Also with the latest commits the solution is not so complex.

Reply:
Sure, but added another comment I think some check has to be done.